portal/server/utils/csv-utils.test.js

```diff
@@ -245,5 +276,49 @@ describe('csv-utils', () => {
 
       expect(extractedValue).toEqual(123);
     });
+
+    it.each([
+      { cellValue: { value: 761579.3699999999 }, expectedValue: 761579.37 },
+      { cellValue: { value: 3938753.8000000007 }, expectedValue: 3938753.8 },
+      { cellValue: { value: 1.230001 }, expectedValue: 1.23 },
+      { cellValue: { value: 1.2300011 }, expectedValue: 1.2300011 },
+    ])('returns $expectedValue for numeric input $cellValue.value and handles potential floating-point issues', async ({ cellValue, expectedValue }) => {
+      const extractedValue = extractCellValue(cellValue);
+
+      expect(extractedValue).toEqual(expectedValue);
+    });
   });
+
+  describe('handleFloatingPointRoundingErrors', () => {
+    it.each([
+      { input: 761579.3699999999, expected: 761579.37 },
+      { input: 3938753.8000000007, expected: 3938753.8 },
+      { input: 9999.9999999999, expected: 10000 },
+      { input: 1.2300009, expected: 1.23 },
+      { input: 1.230001, expected: 1.23 },
+    ])('should return $expected for $input (within floating-point rounding tolerance)', ({ input, expected }) => {
+      expect(handleFloatingPointRoundingErrors(input)).toBe(expected);
+    });
+
+    it.each([123.456, 0.1, 1.2300011, Number.MAX_SAFE_INTEGER])('should return %f unchanged (outside floating-point rounding tolerance)', (value) => {
+      expect(handleFloatingPointRoundingErrors(value)).toBe(value);
+    });
+
+    it.each([456, 1234567])('should return integer %d unchanged', (value) => {
+      expect(handleFloatingPointRoundingErrors(value)).toBe(value);
+    });
+
+    it.each([
+      { input: -761579.3699999999, expected: -761579.37 },
+      { input: -3938753.8000000007, expected: -3938753.8 },
+      { input: -123.456, expected: -123.456 },
+      { input: Number.MIN_SAFE_INTEGER, expected: Number.MIN_SAFE_INTEGER },
+    ])('should return $expected for negative number $input', ({ input, expected }) => {
+      expect(handleFloatingPointRoundingErrors(input)).toBe(expected);
+    });
+
+    it.each(['123.456', null, undefined])('should throw TypeError for non-number input %o', (input) => {
```

Comment:
Nice assertions for type checking, can you please add one with `1e-6` and `1e-5`.

Reply:
Sorry, bit confused - these are valid numbers (and the string case is covered by `123.456`).

If we want to check the boundary, I think these values would be more suitable in the respective tests above:
- `{ input: 9e-7, expected: 0 }` added to the 'inside tolerance' case
- `1e-6` added to the 'outside tolerance' case

Happy to make this change if you're happy with it

Reply:
Added boundary cases to the tests (in line with the latest changes) `987654321.1200005` and `987654321.1200006`.